### PR TITLE
[KT] radix_sort: throw bad_alloc upon memory allocation issues

### PR DESCRIPTION
**Pull request by dmitriy-sobolev**
This follows the same approach as the scan KT and adheres the standard c++ algorithms convention regarding memory allocation issues, for example: "If the algorithm fails to allocate memory, [std::bad_alloc](https://en.cppreference.com/w/cpp/memory/new/bad_alloc) is thrown (see https://en.cppreference.com/w/cpp/algorithm/sort)

### CODE REVIEW

**Review by danhoeflinger — COMMENTED** [review]
Minor wording suggestion, otherwise LGTM.

File: documentation/library_guide/kernel_templates/esimd/radix_sort.rst [hunk 1]
```diff
@@ -219,9 +219,10 @@ Memory Requirements
 
 The algorithm uses global and local device memory (see `SYCL 2020 Specification
 <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_sycl_device_memory_model>`_)
-for intermediate data storage. For the algorithm to operate correctly, there must be enough memory
-on the device; otherwise, the behavior is undefined. The amount of memory that is required
-depends on input data and configuration parameters, as described below.
+for intermediate data storage. For the algorithm to operate correctly, there must be enough memory on the device.
+It throws a ``std::bad_alloc`` exception if there is not enough global device memory.
```
[the remark this answers is not in the document]
Comment (dmitriy-sobolev) [reply documentation/library_guide/kernel_templates/esimd/radix_sort.rst]: Done. I've also applied it to scan KT.

File: documentation/library_guide/kernel_templates/esimd/radix_sort_by_key.rst [hunk 2]
```diff
@@ -264,9 +264,10 @@ Memory Requirements
 
 The algorithm uses global and local device memory (see `SYCL 2020 Specification
 <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_sycl_device_memory_model>`_)
-for intermediate data storage. For the algorithm to operate correctly, there must be enough memory
-on the device; otherwise, the behavior is undefined. The amount of memory that is required
-depends on input data and configuration parameters, as described below.
+for intermediate data storage. For the algorithm to operate correctly, there must be enough memory on the device.
+It throws a ``std::bad_alloc`` exception if there is not enough global device memory.
```
[the remark this answers is not in the document]
Comment (dmitriy-sobolev) [reply documentation/library_guide/kernel_templates/esimd/radix_sort_by_key.rst]: Done. I've also applied it to scan KT.

**Review by danhoeflinger — APPROVED** [review]
LGTM

**Review by adamfidel — APPROVED** [review]
LGTM.

**Pull request merged**